app/enterprise/1.5.x/deployment/dns-considerations.md

```diff
@@ -0,0 +1,258 @@
+proxy in front of the applications (because it uses HTTP path-based routing).
+The Kong proxy can be used for this. The second option requires more
+configuration in kong.conf, but can be used without proxying the applications.
+
+## CORS
+
+### Understanding CORS
+
+[Cross-Origin Resource Sharing, or CORS][mdn-cors], is a set of rules for web
+applications that make requests across origins, i.e.  to URLs that do not share
+the same scheme, hostname, and port as the page making the request. When making
+a cross-origin request, browsers send an `Origin` request header, and servers
+must respond with a matching `Access-Control-Allow-Origin` (ACAO) header. If
+the two headers do not match, the browser will discard the response, and any
+application components that require that response's data will not function
+properly.
+
+For example, the following request/response pairs have matching CORS headers,
+and will succeed:
+
+```
+GET / HTTP/1.1
+Host: example.com
+Origin: http://example.net
+
+HTTP/1.1 200 OK
+Access-Control-Allow-Origin: http://example.net
+```
+
+```
+GET / HTTP/1.1
+Host: example.com
+Origin: http://example.net
+
+HTTP/1.1 200 OK
+Access-Control-Allow-Origin: *
+```
+
+`*` indicates that any origin is allowed.
+
+These requests do not have a matching pair of CORS headers, and will fail:
+
+```
+GET / HTTP/1.1
+Host: example.com
+Origin: http://example.net
+
+HTTP/1.1 200 OK
+Access-Control-Allow-Origin: http://badbadcors.example
+```
+
+```
+GET / HTTP/1.1
+Host: example.com
+Origin: http://example.net
+
+HTTP/1.1 200 OK
+```
+
+Missing CORS headers when CORS headers are expected results in failure.
+
+### CORS in the context of Kong Enterprise
+
+Kong Manager and the Dev Portal operate by issuing requests to their respective
+APIs using JavaScript. These requests may be cross-origin depending on your
+environment.
+
+Kong's services determine what CORS headers to send based on various location
+hint settings in `kong.conf`. The Admin API obtains its ACAO header value from
+`admin_gui_url` and the Portal API obtains its header value from the
+information in the `portal_gui_protocol`, `portal_gui_host`, and
+`portal_gui_use_subdomains` settings. You may optionally specify additional
+Portal API origins using `portal_cors_origins`.
+
+You can configure your environment such that these requests are not
+cross-origin by accessing both the GUI and its associated API via the same
+hostname, e.g. by accessing Kong Manager at `https://admin.kong.example/` and
+the Admin API at `https://admin.kong.example/_api/`. This option requires
+placing a proxy in front of both Kong Manager and the Admin API to handle
+path-based routing; you can use Kong's proxy for this purpose. Note that the
+GUIs must be served at the root of their domains; you cannot place the APIs at
+the root and the GUI under a path.
+
+### Troubleshooting
+
+CORS errors are shown in the browser developer console (see documentation for
```

Comment:
```suggestion
CORS errors are shown in the browser developer console (for example, see documentation for
```
Adding this in case of users in other browsers. Don't want to make it sound like these are the only options.